### PR TITLE
Pm notify channel

### PR DESCRIPTION
**Pull request by fyin1**
Refine the guest power event notification related code a little bit. To make it easy that add vuart as new channel.

Tracked-on: #3564
Signed-off-by: Yin Fengwei <fengwei.yin@intel.com>
Reviewed-by: Yuan Liu <yuan1.liu@intel.com>
Acked-by: Wang Yu <yu1.wang@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by fyin1** [comment]
Wenling,
Sorry. This patch need launch script changed as well. Should I update launch script in this patchset as well?

**Comment by wenlingz** [comment]
> Wenling,
> Sorry. This patch need launch script changed as well. Should I update launch script in this patchset as well?

OK

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: devicemodel/core/main.c [hunk 1]
```diff
@@ -137,7 +138,7 @@ usage(int code)
 		"       %*s [--part_info part_info_name] [--enable_trusty] [--intr_monitor param_setting]\n"
 		"       %*s [--vtpm2 sock_path] [--virtio_poll interval] [--mac_seed seed_string]\n"
 		"       %*s [--vmcfg sub_options] [--dump vm_idx] [--ptdev_no_reset] [--debugexit] \n"
-		"       %*s [--logger-setting param_setting] <vm>\n"
+		"       %*s [--logger-setting param_setting] [--guest_notify_method method] <vm>\n"
```
Comment (gvancuts) [inline devicemodel/core/main.c]: Is this correct? (you use `--pm_notify_channel` in the rest of this patchset)?

Comment (gvancuts) [inline devicemodel/core/main.c]: Is this correct? (you use `--pm_notify_channel` in the rest of this patchset)?

File: devicemodel/core/main.c [hunk 2]
```diff
@@ -171,7 +172,8 @@ usage(int code)
 		"       --vtpm2: Virtual TPM2 args: sock_path=$PATH_OF_SWTPM_SOCKET\n"
 		"       --lapic_pt: enable local apic passthrough\n"
 		"       --rtvm: indicate that the guest is rtvm\n"
-		"       --logger_setting: params like console,level=4;kmsg,level=3\n",
+		"       --logger_setting: params like console,level=4;kmsg,level=3\n"
+		"       --pm_notify_channel: define the channel used to notify guest about power event\n",
```
Comment (gvancuts) [inline devicemodel/core/main.c]: Can you also update `doc/user-guides/acrn-dm-parameters.rst` please?

File: devicemodel/core/main.c (hunk 1 above)
Comment (fyin1) [reply devicemodel/core/main.c]: Yes.

File: devicemodel/core/main.c (hunk 2 above)
Comment (fyin1) [reply devicemodel/core/main.c]: Geoff, sure. The document update is on our todo list. There will be one more patchset will add new parameter of DM. Is it OK that we update the dm parameters document after that patchset is merged?

File: devicemodel/core/main.c (hunk 1 above)
Comment (gvancuts) [reply devicemodel/core/main.c]: OK, I'm not understanding this then. How do you use this `--guest_notify_method` parameter?

File: devicemodel/core/main.c (hunk 2 above)
Comment (gvancuts) [reply devicemodel/core/main.c]: I'd prefer to do this directly in this patch (given it's such a trivial doc change) but OK to do it later.

File: devicemodel/core/main.c (hunk 1 above)
Comment (fyin1) [reply devicemodel/core/main.c]: Oh. Just get your point. This guest_notify_method is incorrect. It should be "--pm_notify_channel".
Thanks a lot for pointing this out.

Comment (fyin1) [reply devicemodel/core/main.c]: Will update the patch after fix failure reported by CI.

File: devicemodel/core/main.c (hunk 2 above)
Comment (fyin1) [reply devicemodel/core/main.c]: OK. Will update the document when I create new PR.

**Comment by fyin1** [comment]
The new patchset only correct github issue info and add document about new parameter.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: devicemodel/core/main.c (hunk 2 above)
Comment (gvancuts) [reply devicemodel/core/main.c]: Thanks!

File: doc/user-guides/acrn-dm-parameters.rst [hunk 3]
```diff
@@ -317,3 +317,14 @@ Here are descriptions for each of these ``acrn-dm`` command line parameters:
        running. It can only be powered off from inside the VM itself.
 
        By default, this option is not enabled.
+
+   * - :kbd:`--pm_notify_channel`
+     - This option is used to define which channel could be used DM to
+       communicate with VM about power management event.
+
+       ACRN supports two channels: ioc and power button.
```
Comment (gvancuts) [inline doc/user-guides/acrn-dm-parameters.rst]:
```suggestion
       ACRN supports two channels: ``ioc`` and ``power_button``.
```

File: doc/user-guides/acrn-dm-parameters.rst [hunk 4]
```diff
@@ -317,3 +317,14 @@ Here are descriptions for each of these ``acrn-dm`` command line parameters:
        running. It can only be powered off from inside the VM itself.
 
        By default, this option is not enabled.
+
+   * - :kbd:`--pm_notify_channel`
```
Comment (gvancuts) [inline doc/user-guides/acrn-dm-parameters.rst]:
```suggestion
   * - :kbd:`--pm_notify_channel <channel>`
```

Comment (fyin1) [reply doc/user-guides/acrn-dm-parameters.rst]: Geoff, thanks a lot for the suggestions. I will update the PR accordingly.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by fyin1** [comment]
Rebase to latest code base.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**